### PR TITLE
[PVR] Fixes for hidden channels

### PR DESCRIPTION
**Pull request by ksooo**
Fixes two small issues with hidden channels, reported in the forum: https://forum.kodi.tv/showthread.php?tid=333402

Runtime tested on macOS, latest Kodi master.

@Jalle19 if you are around, mind taking a look?

### CODE REVIEW

File: xbmc/pvr/channels/PVRChannelGroup.cpp [hunk 1]
```diff
@@ -498,9 +498,10 @@ int CPVRChannelGroup::GetMembers(CFileItemList &results, bool bGroupMembers /* =
 
   for (PVR_CHANNEL_GROUP_SORTED_MEMBERS::const_iterator it = channels->m_sortedMembers.begin(); it != channels->m_sortedMembers.end(); ++it)
   {
-    if (bGroupMembers || !IsGroupMember((*it).channel))
+    const CPVRChannelPtr channel = (*it).channel;
```
Comment (Rechi) [inline xbmc/pvr/channels/PVRChannelGroup.cpp]: replace the for loop with a range based for loop

File: xbmc/pvr/channels/PVRChannelGroup.cpp [hunk 2]
```diff
@@ -498,9 +498,10 @@ int CPVRChannelGroup::GetMembers(CFileItemList &results, bool bGroupMembers /* =
...
+    if (bGroupMembers != channel->IsHidden() || !IsGroupMember(channel))
```
Comment (Jalle19) [inline xbmc/pvr/channels/PVRChannelGroup.cpp]: This logic is a bit hard to understand, a longer more expressive boolean statement would help

File: xbmc/pvr/channels/PVRChannelGroup.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/pvr/channels/PVRChannelGroup.cpp]: not in scope of this PR

File: xbmc/pvr/channels/PVRChannelGroup.cpp (hunk 2 above)
Comment (ksooo) [reply xbmc/pvr/channels/PVRChannelGroup.cpp]: Agreed, took me a while to understand this as well. Will see how I can improve it to get more readable.

File: xbmc/pvr/channels/PVRChannelGroup.cpp (hunk 1 above)
Comment (Rechi) [reply xbmc/pvr/channels/PVRChannelGroup.cpp]: This is already a non functional change, then why not do it correct?

**Comment by ksooo** [comment]
@Jalle19 cleaned up the mess. Should be more readable now (not the diff, but the code ;-)

**Review by Jalle19 — APPROVED** [review]

**Pull request merged**